compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -380,10 +399,32 @@ private boolean containsAnyType(BType type) {
         return false;
     }
 
+    private boolean containsAnyDataType(BType type) {
+        if (type.tag != TypeTags.UNION) {
+            return type.tag == TypeTags.ANYDATA;
+        }
+
+        for (BType memberTypes : ((BUnionType) type).getMemberTypes()) {
+            if (memberTypes.tag == TypeTags.ANYDATA) {
+                return true;
+            }
+        }
+        return false;
+    }
+
     public BType mergeTypes(BType typeFirst, BType typeSecond) {
         if (containsAnyType(typeFirst)) {
             return typeSecond;
         }
+        if (containsAnyType(typeSecond)) {
+            return typeFirst;
+        }
+        if (containsAnyDataType(typeFirst)) {
+            return typeSecond;
+        }
+        if (containsAnyDataType(typeSecond)) {
+            return typeFirst;
+        }
```

Comment:
Neither `any` nor `anydata` contain `error`. So shouldn't we check what the other type is here?

Reply:
Yes, fixed